src/main/groovy/com/palantir/gradle/docker/DockerComposeUp.groovy

```diff
@@ -0,0 +1,36 @@
+package com.palantir.gradle.docker
+
+import groovy.util.logging.Slf4j
+import org.gradle.api.DefaultTask
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.TaskAction
+
+@Slf4j
+class DockerComposeUp extends DefaultTask {
```

Comment:
Could we extend from an [Exec](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.Exec.html) task so that people can configure/override more stuff?

Reply:
Is there best practices regarding defining a task in a separate groovy class vs directly in the plugin class? We do the latter in `PalantirDockerPlugin` although a separate class makes things cleaner.

Reply:
Separate class is generally the way to go as it's much easier to add clear annotations for incremental behaviour, description etc etc.

It's also much easier to convert the separate classes from Groovy -> Java and gradle recommend writing plugins in Java over groovy https://docs.gradle.org/current/userguide/custom_plugins.html

Reply:
Best practice is to not extend Exec.

Reply:
OK fair enough, this will probably be fine as an MVP anyway - can always add more configurability if people really need it later

Reply:
Yeah - I would say composition with `project.exec` provides way more flexibility than directly extending `Exec` since you're bound to execs `TaskAction` method.